homeworks/07-mars-viewer/src/components/imagesBlock/ImagesBlock.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { MouseEventHandler, useCallback } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { types } from '../../store/actions/actionsTypes';
+import { RootState } from '../../store/reducers/rootReducer';
+import style from '../imagesContainer/imagesContainer.module.css';
+
+export interface CurImg {
+  curImg: string
+  name: string
+  curId: string,
+  active: boolean
+}
+
+export const ImagesBlock: React.FC<CurImg> = ({
+  // eslint-disable-next-line react/prop-types
+  curImg, name, curId, active,
+}) => {
+  const { sols } = useSelector((state: RootState) => state.fetchSols);
+  const { favourites } = useSelector((state: RootState) => state.favouritesSols);
+  const dispatch = useDispatch();
+  const handleClickSvg: MouseEventHandler<SVGSVGElement> = useCallback((event) => {
+    const { id } = event.currentTarget;
+    const result = favourites.filter((favourite) => +favourite === +id).length > 0;
+    if (!result) {
+      dispatch({
+        type: types.AddFavourite,
+        payload: { favourite: id },
+      });
+    } else {
+      dispatch({
+        type: types.DeleteFavourite,
+        payload: { favourite: id },
+      });
+    }
+  }, [dispatch, sols]);
+  return (
+    <li
+      key={Math.random()}
```

Comment:
pass consistent value, this will lead to unmount every render, learn more here https://ru.reactjs.org/docs/lists-and-keys.html